File: graphql-api/src/main/java/io/prizy/publicapi/application/Application.java
package io.prizy.publicapi.application;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

/**
 * @author Nidhal Dogga
 * @created 6/24/2022 7:23 PM
 */


@SpringBootApplication
@ComponentScan(basePackages = "io.prizy.publicapi")
public class Application {

  public static void main(String[] args) {
    SpringApplication.run(Application.class, args);
  }

}

File: public-api/src/main/kotlin/io/prizy/publicapi/application/Application.kt
package io.prizy.publicapi.application

import io.prizy.publicapi.application.properties.AppConfigurationProperties
import io.prizy.publicapi.application.properties.GameProperties
import io.prizy.publicapi.application.properties.ResourceProperties
import io.prizy.publicapi.application.properties.StorageProperties
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication
import org.springframework.context.annotation.ComponentScan
import org.springframework.scheduling.annotation.EnableAsync

@EnableAsync
@SpringBootApplication
@ComponentScan(basePackages = ["io.prizy.publicapi"])
@EnableConfigurationProperties(
  GameProperties::class,
  StorageProperties::class,
  ResourceProperties::class,
  AppConfigurationProperties::class
)
class Application

fun main(args: Array<String>) {
  runApplication<Application>(*args)
}
